### ISSUE TITLE
ON HOLD: Adaptive factor values

### ISSUE THREAD

**Issue by npaton** [malsobay/pgg#23]
In the current implementation, the game's logic adapts to however many players are actually connected to the game, and doesn't relay on the playerCount factor to calculate things like payoffs. However, there are situations in which we'd want a factor's value to be conditional on another's. 

For example, say we have a multiplier of 3 for a game with 4 players. In this situation, a player's "return on investment" (ROI) for each coin they contribute is 0.75 (ie 3/4). However, if a player drops out mid-game, the ROI goes up to 1, and the player is indifferent to contributing or withholding their coins — this is a problem if we care about maintaining the same value of ROI. One solution could be for the multiplier factor in the treatment to adjust to the number of actual players in the game, eg when the player disconnected in the example, the multiplier would change to 2.25 to maintain an ROI of 0.75.

### Implementation proposal

onRoundStart, we copy the treatment from the game, pass it into a function that will rework the values, and then set it on the round. Everywhere the treatment is used, it should go from `game.get("treatment")` to `round.get("treatment")`.

**Comment by npaton** [comment]
@malsobay This one is on hold I think, right? Let us know.

**Comment by malsobay** [comment]
I agree that, for now, this is the lowest priority of the three ideas for development that we discussed. 

That said, it is somewhat related to the more general idea of having a framework for treatment validation; such validation would also be useful for the network structure use case, where we would check that dim(adjacency matrix) = playerCount, for example.
